src/CodeCompliance/Domain/Entity/Report.php

```diff
@@ -25,7 +43,7 @@ class Report implements ViolationReportInterface
     protected string $path = '';
 
     /**
-     * @var array<\SprykerSdk\SdkContracts\Report\Violation\ViolationInterface>
+     * @var array<\CodeCompliance\Domain\Entity\Violation>
```

Comment:
It is always better to work with interfaces, not with the class. The most preferable way will be if we can use further the violation interface from the contracts. 

But if for some reason you need to extend the interface, you can create a new interface, that will extend multiple other interfaces, and you can work with it. Example:

```
interface MyAwesomeInterface extends InterfaceA, ArrayAccess, Serializable
{
    ...
}
```

Reply:
In this case, Domain\Entity objects are DTO. Also for DTO we can use class instead of interface - it was my reason for this. 

Of course we can extend interfaces on this level

Reply:
For protocol - as discussed, we can leave it here without changes as long as this DTO object is used only intern in our tool.

Reply:
Added ViolationInterface ReportInterface PackageViolationReportInterface on the Upgrader level